src/lib.rs

```diff
@@ -100,8 +100,10 @@ mod arch {
                 command.args(&c);
             }
             Workload::Pid(p) => {
+                let p = p.into_iter().map(|p| p.to_string()).collect::<Vec<_>>();
```

Comment:
Should be able to avoid the intermediate allocations here.

Reply:
Not a big fan of the new solution, but the `join` method is only for slices of `Borrow<str>`, so I implemented it manually to make it one argument for perf.